### ISSUE TITLE
Removetwisted.python.test.test_util._getDeprecationSuppression.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6266]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)| @tomprince reported|
|-|-|
|Trac ID|trac#6266|
|Type|enhancement|
|Created|2013-01-19 02:57:03Z|
|Branch|https://github.com/twisted/twisted/tree/getDeprecationSuppression-removal-6266|

According to [#5645](https://github.com/twisted/twisted/issues/5645), when deprecating something, the tests of that functionality need to remain, but not emit deprecation warnings.

In [#5544](https://github.com/twisted/twisted/issues/5544), a private function `_getDeprecationSuppression` was added to generate an approriate narrow suppression (to pass to `.suppress` of a test case or method).

This function should be made public.

<details><summary>Searchable metadata</summary>

```
trac-id__6266 6266
type__enhancement enhancement
reporter__tom_prince tom.prince
priority__normal normal
milestone__ 
branch__branches_getDeprecationSuppression_removal_6266 branches/getDeprecationSuppression-removal-6266
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1358564223000000 1358564223000000
changetime__1462461756271350 1462461756271350
version__None None
owner__adiroiban adiroiban

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_1"></a>|@tomprince commented|
|-|-|

(fix ticket link in description)

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban commented|
|-|-|

I think that this API is no longer valid and the whole suppression thing is no longer valid.

I have removed the api to clean the code.

please review.

thanks!

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph set owner to @adiroiban|
|-|-|

> I think that this API is no longer valid and the whole suppression thing is no longer valid.

Just out of curiosity, is there documentation someplace about the right way to do this now, if it's not using suppressions?

> I have removed the api to clean the code.

LGTM.  If you can fix the pyflakes error that would be nice, or file a bug on braid or something if it's spurious.

> please review.

Looks like this is just deleting totally unused and untested code, so: go for it!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban commented|
|-|-|

Thanks for the review.

--------

I hope that this is covered by current documentation here https://github.com/twisted/twisted/blob/trunk/docs/core/development/policy/compatibility-policy.rst#testing-deprecation-code

Not sure if this is the same thing, since the removed code is never used and I am not sure if this is about the same thing.

---------

pyflakes error is valid... pyflakes is most of the time right :)

I hope that soon we will have a twisted helper to run pyflakes on local branches.

-------

will merge

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban set status to `closed`|
|-|-|

(In [47345]) Merge getDeprecationSuppression-removal-6266: Remove t.p.test.test_util._getDeprecationSuppression.

Author: adiroiban
Reviewer: glyph
Fixes: [#6266](https://github.com/twisted/twisted/issues/6266)
